src/Compiler/Microsoft.AspNetCore.Razor.Language/src/Legacy/MarkupChunkGenerator.cs

```diff
@@ -7,6 +7,10 @@ namespace Microsoft.AspNetCore.Razor.Language.Legacy;
 
 internal class MarkupChunkGenerator : SpanChunkGenerator
 {
+    public static readonly MarkupChunkGenerator Instance = new();
+
+    private MarkupChunkGenerator() { }
+
```

Comment:
Did you consider overriding `Equals` and `GetHashCode` as well? The implementations in `SpanChunkGenerator` can be simplified now that this is a singleton. Not a big diff but noticed when looking at the code.

It overall feels a bit odd that every implementation of `SpanChunkGenerator` defines their own equality except for the two that can be singleton's.